tests/kw_config_loader_test.sh

```diff
@@ -121,6 +121,7 @@ function testLocalConfigFile
 function testConfigFilesLoadingOrder
 {
   expected="$etc_files_path/$CONFIG_FILENAME
+$HOME/.kw/$CONFIG_FILENAME
 $PWD/$CONFIG_FILENAME"
```

Comment:
The way that you handle this test is really nice.
I have a different approach to this sort of situation; I tried to isolate the function that I want to change the behavior at the beginning of the test for making the test straightforward. In this case, something like this:

```
diff --git a/tests/kw_config_loader_test.sh b/tests/kw_config_loader_test.sh
index 2a6c754..16ca136 100755
--- a/tests/kw_config_loader_test.sh
+++ b/tests/kw_config_loader_test.sh
@@ -13,8 +13,15 @@ function suite
   suite_addTest "testConfigFilesLoadingOrder"
 }
 
+function mock_parse_configuration()
+{
+  echo "$@"
+}
+
 function setUp
 {
+  shopt -s expand_aliases
+  alias parse_configuration='mock_parse_configuration'
     mkdir -p "$TMP_DIR"
     cp "$PWD/etc/kworkflow_template.config" "$TMP_DIR/kworkflow.config"
 }
@@ -124,10 +131,7 @@ function testConfigFilesLoadingOrder
 $HOME/.kw/$CONFIG_FILENAME
 $PWD/$CONFIG_FILENAME"
 
-  output="$(
-    function parse_configuration { echo "$@"; }
-    load_configuration
-  )"
+  output="$(load_configuration)"
 
   [[ "$output" == "$expected" ]]
   assertTrue "Wrong config file reading order." $?
```

What do you think?

Reply:
I like it, but then, wouldn't `mock_parse_configuration()` be executed -- in place of `parse_configuration()` -- in every test instead of just this one?

Reply:
Hmmmmm... nice catch!
An option could be creating setMockFunction, and just call it inside testConfigFilesLoadingOrder. However, I don't have a strong opinion about it; just let me know what you want to do here.